### PR TITLE
Require cryptosuite specifications to mitigate against poison graphs.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address issue #87 by 1) documenting that some transformations (like RDF Dataset Canonicalization) can be attacked, and 2) that there are mitigations for those attacks and cryptosuite specifications MUST ensure that those mitigations are implemented.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-integrity/pull/128.html" title="Last updated on Jul 29, 2023, 6:08 PM UTC (d6bb627)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-integrity/128/1272e58...d6bb627.html" title="Last updated on Jul 29, 2023, 6:08 PM UTC (d6bb627)">Diff</a>

### CODE REVIEW

**Review by dlongley — APPROVED** [review]

**Review by Wind4Greg — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -2130,6 +2134,23 @@ <h3>Transformations</h3>
 being protected. Developers are also urged to regularly confirm that the
 cryptographically protected data has not been tampered with as it is written to
 and read from application storage.
+        </p>
+        <p class="advisement">
+Some transformations, such as RDF Dataset Canonicalization [[RDF-CANON]], have
+mitigations for input data sets that can be used by attackers to consume
+excessive processing cycles. These class of attacks are called
```
Comment (TallTed) [inline index.html]:
```suggestion
excessive processing cycles. This class of attack is called
```

File: index.html [hunk 2]
```diff
@@ -2130,6 +2134,23 @@ <h3>Transformations</h3>
...
+<a data-cite="RDF-CANON#dataset-poisoning">dataset poisoning</a> and all modern
```
Comment (TallTed) [inline index.html]:
```suggestion
<a data-cite="RDF-CANON#dataset-poisoning">dataset poisoning</a>, and all modern
```

File: index.html [hunk 3]
```diff
@@ -2130,6 +2134,23 @@ <h3>Transformations</h3>
...
+RDF Dataset canonicalizers are required to detect these sorts of bad inputs and
+halt processing. The test suites for RDF Dataset Canonicalization contains such
```
Comment (TallTed) [inline index.html]:
```suggestion
halt processing. The test suites for RDF Dataset Canonicalization includes such
```

File: index.html [hunk 4]
```diff
@@ -2130,6 +2134,23 @@ <h3>Transformations</h3>
...
+poisoned datasets to ensure that such mitigations exist in all conforming
+implementations. Generally speaking, cryptographic suite specifications that
+utilize transformations are required to mitigate these sorts of attacks and
```
Comment (TallTed) [inline index.html]:
```suggestion
use transformations are required to mitigate these sorts of attacks, and
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Review by dmitrizagidulin — APPROVED** [review]
:+1: (once TallTed's changes are applied)

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**